### ISSUE TITLE
jsx-handler-names: false positive when handler name begins with number

### ISSUE THREAD

**Issue by jsphstls** [yannickcr/eslint-plugin-react#1686]
![image](https://user-images.githubusercontent.com/4481784/36049520-4164c64c-0da0-11e8-8205-e7f459aff9b9.png)

Fail:
```javascript
onClick={this.handle123LogoClick}
```

Pass
```javascript
onClick={this.handleLogo123Click}
```

123Logo is a fictional example.

**Comment by jsphstls** [comment]
Pull request: https://github.com/yannickcr/eslint-plugin-react/pull/1689

**Comment by Moong0122** [comment]
May I try to solve this bug?

**Comment by ljharb** [comment]
@Moong0122 any issue with a "help wanted" label that doesn't have a linked PR is up for grabs. This one has #1689, though (which I'll take another look at) so i don't think any additional help is needed on this issue.
